sql/src/main/java/org/opensearch/sql/sql/antlr/SQLSyntaxParser.java

```diff
@@ -19,17 +21,24 @@
  */
 public class SQLSyntaxParser implements Parser {
 
+  private static final Logger LOG = LogManager.getLogger(SQLSyntaxParser.class);
+
   /**
    * Parse a SQL query by ANTLR parser.
    * @param query   a SQL query
    * @return        parse tree root
    */
   @Override
   public ParseTree parse(String query) {
+    AnonymizerListener anonymizer = new AnonymizerListener();
+
     OpenSearchSQLLexer lexer = new OpenSearchSQLLexer(new CaseInsensitiveCharStream(query));
     OpenSearchSQLParser parser = new OpenSearchSQLParser(new CommonTokenStream(lexer));
     parser.addErrorListener(new SyntaxAnalysisErrorListener());
-    return parser.root();
-  }
+    parser.addParseListener(anonymizer);
```

Comment:
```suggestion
    parser.addParseListener(new AnonymizerListener());
```

Reply:
I need the anonymizer listener for line 41 where I call the `getAnonymizedQueryString()` from it so sadly I don't believe that I can simplify it like this